egg/core/trainers.py

```diff
@@ -196,6 +196,11 @@ def train(self, n_epochs):
                     callback.on_test_end(validation_loss, validation_interaction, epoch+1)
 
             if self.should_stop:
+                for callback in self.callbacks:
+                    if not ('validation_loss' in locals() or 'validation_interaction' in locals()):
+                        validation_loss = validation_interaction = None
```

Comment:
q: would things break if we just set (validation_loss,  validation_interaction) to None say on L189? Then no need for this check, they'd be either None or real values already